### PR TITLE
[DOP-22108] Rename backend to server

### PR DESCRIPTION
**Pull request by IlyasDevelopment**
## Change Summary

- Renamed `backend` to `server` across the codebase, Docker configuration, and documentation:  
  - **Code**: Updated `syncmaster.backend` to `syncmaster.server`.  
  - **Docker**:  
    - Renamed Docker image from `syncmaster-backend` to `syncmaster-server`.  
    - Updated `Dockerfile` and `docker-compose.yml` to reflect the new naming.  
  - **Documentation**: Renamed `backend/index.rst` to `server/index.rst`.  

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [x] Unit and integration tests for the changes exist
* [x] Tests pass on CI and coverage does not decrease
* [ ] Documentation reflects the changes where applicable
* [ ] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/syncmaster/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: syncmaster/server/__init__.py [hunk 1]
```diff
@@ -6,26 +6,26 @@
...
+from syncmaster.server.services.unit_of_work import UnitOfWork
+from syncmaster.server.settings import ServerAppSettings as Settings
 
 
 def celery_factory(settings: Settings) -> Celery:
     app = Celery(
         __name__,
         broker=settings.broker.url,
-        backend="db+" + settings.database.sync_url,
+        server="db+" + settings.database.sync_url,
```
Comment (dolfinus) [inline syncmaster/server/__init__.py:28]: Please rever this line, and same for other celery_factory implementations

File: syncmaster/server/settings/server/__init__.py [hunk 2]
```diff
@@ -44,21 +44,21 @@ class ServerSettings(BaseModel):
     )
     session: SessionSettings = Field(
         default_factory=SessionSettings,  # type: ignore[arg-type]
-        description=":ref:`Session settings <backend-configuration-server-session>`",
+        description=":ref:`Session settings <server-configuration-server-session>`",
```
Comment (dolfinus) [inline syncmaster/server/settings/server/__init__.py:47]: Well, `backend-configuration-` prefix is probably okay for now, as we have different settings subsections for server, scheduler, worker and so on. If not, then we should replace `server-configuration-server-` with just `server-configuration-`

File: Makefile [hunk 3]
```diff
@@ -5,7 +5,7 @@ include .env.local
 PIP = .venv/bin/pip
 POETRY = .venv/bin/poetry
 
-# Fix docker build and docker compose build using different backends
+# Fix docker build and docker compose build using different servers
```
Comment (dolfinus) [inline Makefile:8]: Please revers this line

File: docs/scheduler/configuration/logging.rst [hunk 4]
```diff
@@ -1,4 +1,4 @@
-.. _backend-configuration-logging:
+.. _server-configuration-logging:
```
Comment (dolfinus) [inline docs/scheduler/configuration/logging.rst:1]: This file is in `docs/scheduler` folder, not `docs/server`. Probably current prefix `backend-configuration` is okay, as this is used in all backend components (server, scheduler, worker)

File: docs/scheduler/start_scheduler.rst [hunk 5]
```diff
@@ -7,7 +7,7 @@ With docker
 Installation process
 ~~~~~~~~~~~~~~~~~~~~
 
-Docker will download worker image of syncmaster scheduler (same as backend image) & broker, and run them.
+Docker will download worker image of syncmaster scheduler (same as server image) & broker, and run them.
```
Comment (dolfinus) [inline docs/scheduler/start_scheduler.rst:10]:
```suggestion
Docker will download worker image of syncmaster scheduler & broker, and run them.
```

File: syncmaster/server/settings/server/__init__.py (hunk 2 above)
Comment (dolfinus) [reply syncmaster/server/settings/server/__init__.py]: Actually, now configuration links are inconsistent. There are `server-auth-providers`, `server-install`, `server-openapi`, but `backend-conguration-server-openapi`, `backend-configuration-server-session` and so on.

Probably we should use `server-configuration-openapi`, `server-configuration-session` instead.

**Review by dolfinus — APPROVED** [review]

File: syncmaster/server/settings/server/__init__.py (hunk 2 above)
Comment (dolfinus) [reply syncmaster/server/settings/server/__init__.py]: @IlyasDevelopment what do you think?

**Review by maxim-lixakov — APPROVED** [review]

File: syncmaster/server/settings/server/__init__.py (hunk 2 above)
Comment (IlyasDevelopment) [reply syncmaster/server/settings/server/__init__.py]: Fixed. Now it looks more consistent

**Pull request merged**